python/ray/tests/test_runtime_context.py

```diff
@@ -243,10 +244,16 @@ def test_ids(ray_start_regular):
     rtc = ray.get_runtime_context()
     # node id
     assert isinstance(rtc.get_node_id(), str)
-    assert rtc.get_node_id() == rtc.node_id.hex()
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Hmm this test is a bit weird? What's the meaning of checking this? (maybe we should just test get_* API only without checking if it prints warning?)

Reply:
hmm it does check if calling `rtc.node_id.hex()` send warning message? this is suggested way of test warning:
https://docs.python.org/3/library/warnings.html#testing-warnings